src/suggestions/suggester.py

```diff
@@ -0,0 +1,45 @@
+from typing import Optional
+import re
+
+from config import Rule
+
+from . import SuggestedChange, SuggestedChanges
+
+
+
+class Suggester:
+	@classmethod
+	def load_suggestions(cls, rule: Rule) -> None:
+		if (suggestions := rule.get('suggestions')) is not None:
+			for suggestion in suggestions:
+				suggestion['pattern_regex'] = re.compile(suggestion['pattern'])
+
+	def get_suggestion(self, text: str, rule: Rule) -> str:
+		suggestion = self.suggest(text, rule)
+		if suggestion is None:
+			return ""
+		else:
+			return f"\n\n```suggestion\n{suggestion.suggestion}\n```"
+
+	def suggest(self, text: str, rule: Rule) -> Optional[SuggestedChange]:
+		suggestions = rule.get('suggestions')
+		if suggestions is None:
+			return None
+
+		is_change_made = False
+		suggestion = text
+		while True:
+			found_matches = False
+			for suggestion_config in suggestions:
+				p = suggestion_config['pattern_regex']
+				while p.match(suggestion):
```

Comment:
`match` matches from the beginning of the string, but `sub` matches anywhere in the string.  You may want to use `search` here instead, which also matches anywhere in the string.

Reply:
Good point. I think you're right. I think the pattern should normally start with `^`, so it shouldn't matter, but I'll think more about this and make some tests to see what we need.

Reply:
I'll use subn so that we don't have to check for matches twice.

Reply:
I changed this to use `subn`. So it should be faster now and it lets us have more clear rules, like this one to remove trailing whitespace:
```yaml
    suggestions:
      - pattern: '\s+$',
        replacement: ''
```